lib/pbench/test/unit/server/query_apis/test_datasets_detail.py

```diff
@@ -19,43 +23,45 @@ class TestDatasetsDetail(Commons):
     def _setup(self, client):
         super()._setup(
             cls_obj=DatasetsDetail(client.config, client.logger),
-            pbench_endpoint="/datasets/detail/fio",
+            pbench_endpoint="/datasets/detail/random_md5_string1",
             elastic_endpoint="/_search?ignore_unavailable=true",
-            payload={
-                "user": "drb",
-                "access": "private",
-                "start": "2020-08",
-                "end": "2020-10",
-            },
+            index_from_metadata="run-data",
             empty_es_response_payload=self.EMPTY_ES_RESPONSE_PAYLOAD,
         )
 
+    api_method = API_METHOD.GET
+
+    def token(self, client, config: PbenchServerConfig, user: str) -> str:
+        response = client.post(
+            f"{config.rest_uri}/login",
+            json={"username": user, "password": "12345"},
+        )
+        assert response.status_code == HTTPStatus.OK
+        data = response.json
+        assert data["auth_token"]
+        return data["auth_token"]
+
     @pytest.mark.parametrize(
-        "user",
-        ("drb", "badwolf", "no_user"),
+        "user, expected_status",
+        [
+            ("drb", HTTPStatus.OK),
+            ("test", HTTPStatus.FORBIDDEN),
+            (None, HTTPStatus.UNAUTHORIZED),
+        ],
     )
     def test_query(
-        self, client, server_config, query_api, find_template, build_auth_header, user
+        self, client, server_config, query_api, find_template, user, expected_status
     ):
```

Comment:
The problem with replacing the use of `build_auth_header` is that we're no longer testing ADMIN access.  See [Dave's reply](https://github.com/distributed-system-analysis/pbench/pull/3030#discussion_r989415392) for the remedy.